### ISSUE TITLE
Troubleshooting insightparams subscriptions for Home Assistant

### ISSUE THREAD

**Issue by deftdawg** [pywemo/pywemo#64]
I'm not really a python guy, but I'm trying to troubleshoot why Home Assistant's Wemo switch component does not seem to be receiving insightparam update events (https://github.com/home-assistant/home-assistant/issues/3506)... 

It is able to get insightparams when the device is first discovered when home assistant starts. 

I was able to validate that the Wemo Insight is working by testing against the npm wemo-client package with the following program (updates come about 1 per second):
```
var Wemo = require('wemo-client');
var wemo = new Wemo();
const util = require('util')

wemo.discover(function(deviceInfo) {
  console.log('Wemo Device Found: %j', deviceInfo);
  // Get the client for the found device
  var client = wemo.client(deviceInfo);

  // Handle BinaryState events
  client.on('insightParams', function (binaryState, instantPower, data) {
    console.log('insightParams changed to: %s %s', instantPower, util.inspect(data, {showHidden: false, depth: null}));
  });
});
```

However I've been unable to find a simple subscriptionRegistry example in Python I can use to validate that pywemo is capturing the insightParams events.   This is as far as I got and I'm obviously missing something because I'm not getting any log messages after it says `INFO:pywemo.subscribe:Listening on port 8990` (I probably need an event handler, but not sure how to add it)

```
#!/usr/bin/python3

import sys
import pywemo
import logging
devices = pywemo.discover_devices()
print(devices)
logging.basicConfig(level=logging.DEBUG)
SUBSCRIPTION_REGISTRY = pywemo.SubscriptionRegistry()
SUBSCRIPTION_REGISTRY.start()
```

Hoping someone can point me in the right direction.  Thanks.

**Comment by pavoni** [comment]
This sounds a bit like this issue
https://github.com/home-assistant/home-assistant/issues/3506

I suspect the the wemo device only triggers a callback when something `significant` happens - such as a device turning on / off - or power useage crossing the threshold.

I have a coffee machine that I detect a power useage change when the thermostat triggers (and the power useage crosses the threshold) - but I don't think it sends other messages based on power useage.

Well worth investigating more though!

To get your subscription example to work you need to do something like this:-

https://github.com/home-assistant/home-assistant/blob/dev/homeassistant/components/switch/wemo.py#L60

Also please note that at this point you have to call the update function to fetch state after you get a callback, we don't yet extract state from the callback message.

**Comment by deftdawg** [comment]
> I have a coffee machine that I detect a power useage change when the thermostat triggers - but I don't think it sends other messages based on power useage.

You should run the nodejs program above and see if you get anything after discovery, it's early enough for a second cup. ;)

> I suspect the the wemo device only triggers a callback when something significant happens - such as a device turning on / off - or power useage crossing the threshold.

I suspect the threshold value is a cosmetic thing for users to look at the app and see if something is on (vs. phantom/standby load).  In my set-up, I'm always over (14W) the threshold (8W)...  Since node can see the events coming every sec, I would expect HASS to see them too if it was properly subscribed to that...  I'm starting to wonder if the nodejs implementation is actually polling the Insight for the data...  maybe I'll slap wireshark on it later and see what it's doing.

**Comment by pavoni** [comment]
Afraid I'm not in the same place as my Insight, so can't try right now.

It was a while back that I tested - but I think the threshold is used to trigger an event - try and change it to over 14 and see what happens.

As well as the actual power usage - the wemo reports a `standby` state depending on the threshold.

**Comment by pavoni** [comment]
Looking quickly at the node code - it may be subscribting to another service (not just basicevent).

https://github.com/pavoni/pywemo/blob/master/pywemo/subscribe.py#L113

https://github.com/timonreinhard/wemo-client/blob/master/client.js#L47

**Comment by deftdawg** [comment]
Yeah I noticed the .basicevent.serviceURL line there, but couldn't find a source...  it finally dawned on me (after some sleep) that it might be a reflection, and sure enough it was... 

Anyway, I've hacked the subscribe.py to give the 'insight' subscription priority over 'basicevent' when 'insight' service is present.   With that change I'm getting insight updates every second in Home Assistant. 

The behaviour I observe when testing is that if the switch is on, the insight will publish power stats.

All the rest of my non-insight switches behave normally... It looks like the Insight sub picks up basic on/off events as part of it's sub...
